### ISSUE TITLE
[Feature] Automatic Update

### ISSUE THREAD

**Issue by ahbanavi** [wesmoncrief/obsidian-todoist-text#28]
I use Todoist to quickly jot down ideas on the go using my mobile device.  
At the end of each day, I go through my "On the Go Ideas" project and manually create a note for each task.  
This plugin allows me to keep all of my Todoist ideas in a single Obsidian note, but the current process of manually removing and inserting keywords to update my notes is time-consuming.  
I would greatly appreciate it if the developer could consider implementing an automatic update feature to streamline this workflow.

**Comment by wesmoncrief** [comment]
To clarify, it sounds like there's two distinct things you're talking about.
1. You currently have to manually a note per each task in your 'on the go' project
2. you want to automatically have each of those notes refresh as new tasks are added?

**Comment by ahbanavi** [comment]
> To clarify, it sounds like there's two distinct things you're talking about.
> 1. You currently have to manually a note per each task in your 'on the go' project
> 2. you want to automatically have each of those notes refresh as new tasks are added?

Adding note for each task is not something that intent for this Plugin and that something I should do manually.

I have single note with my ideas from todoist and want that note to be updated at least once or twice a day, right now I have to delete the content and add the keyword to update that manually, that's the part I ment for automation.
![Screenshot_20230113-130342_Obsidian.jpg](https://user-images.githubusercontent.com/39927312/212287107-c5cf7d41-c9a1-4694-8f1e-e97b6cc24887.jpg)
I want only this note to updated, not others that I link manually. 

Also if the auto update feature has the ability to not delete already done (~~strikethrough~~ texts) tasks from obsidian, would be great for archiving.

**Comment by wesmoncrief** [comment]
Makes sense! Similar idea to https://github.com/wesmoncrief/obsidian-todoist-text/issues/17

This would probably be at the top of the priority list if i ever get around to another PR here. Can't promise any specific deliverable time, though if you or anyone else wants to try coding it, i'm happy to review/provide guidance.

I wonder if as a (partial) workaround, you could do something like:
```md

# active things
@@queryActiveTasks@@ 

# finished tasks
@@queryIsCompleteTasks@@ 
```

Since it looks like the filter API supports retrieving completed tasks, you can presumably get historical data that way. You could also perhaps use a Template from the templates plugin to help auto-generate new notes.

I realize that doesn't really solve it in the proper way though

**Comment by ahbanavi** [comment]
I didn't know that filtered API supports completed tasks. That workaround will do the job for me for now.  
Thanks for the advice and also for this plugin.

**Comment by gdicker1** [comment]
@ahbanavi I don't know how helpful this might be maybe just a few keystrokes. But maybe you could add all of your Todoist-text keywords for this file to a Template (from the core plugin).  Then you could do something like: "Select All", "Delete", "CMD+P" (command palette), "Templates: Insert template", and select the file containing your desired Todoist-text keywords.
